source/extensions/filters/network/istio_authn/config.cc

```diff
@@ -28,39 +26,63 @@ namespace Extensions {
 namespace NetworkFilters {
 namespace IstioAuthn {
 
-absl::optional<uint64_t> PeerPrincipal::hash() const {
+absl::optional<uint64_t> Principal::hash() const {
```

Comment:
There are a few terms in this PR related to principal, e.g., Principal, PeerPrincipal (e.g., PeerPrincipalKey), LocalPrincipal (e.g., LocalPrincipalKey).

- Why change PeerPrincipal to Principal? 
- Are you generalizing PeerPrincipal into Principal? If so, will PeerPrincipalKey and LocalPrincipalKey be factored too? 
- Are there any comments in the code describing the differences between Principal, PeerPrincipal, LocalPrincipal?

Reply:
Principal is a shared representation for peer and local principal. The keys refers to peer vs local, but both values are Principal structs. The reason why local is important is multi-tenancy (which we don't have enabled yet), if the proxy can present multiple certificates, we need to detect and propagates the local principal as well.

The refactor was needed because I didn't have local one before, and then realized it's important and needed.